etcd_helper/tests/etcd_test.go

```diff
@@ -0,0 +1,80 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/danielpygo/kubeplus/etcd_helper"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	rdr  etcd_helper.Etcdreader
+	wrtr etcd_helper.Etcdwriter
+)
+
+func init() {
+	rdr.EtcdServiceURL = "http://localhost:2379"
+	wrtr.EtcdServiceURL = "http://localhost:2379"
+}
+
+func TestGet(t *testing.T) {
+	err := wrtr.Store("postgres", "THIS IS SOME DATA")
+	if err != nil {
+		fmt.Printf("Err: %s\n", err)
+	}
+	str, err := rdr.Get("postgres")
+	if err != nil {
+		fmt.Printf("Err: %s\n", err)
+	}
+	assert.Equal(t, str, "THIS IS SOME DATA", "A simple Post and Get of one item")
+}
+func TestGetList(t *testing.T) {
+	dataList := []string{"database1", "database2", "database3"}
+	err := wrtr.StoreList("postgres1", dataList)
+	if err != nil {
+		fmt.Printf("Err: %s\n", err)
+	}
+	strSlice, err := rdr.GetList("postgres1")
+	if err != nil {
+		fmt.Printf("Err: %s\n", err)
+	}
+	assert.Equal(t, strSlice, dataList, "A simple Post and Get of an array of dataItems")
+}
+
+func TestDeleteFromList(t *testing.T) {
+	dataList := []string{"database1", "database2", "database3"}
+	err := wrtr.StoreList("postgres2", dataList)
+	if err != nil {
+		fmt.Printf("Err: %s\n", err)
+	}
+	err = wrtr.Delete("postgres2", "database3")
+	if err != nil {
+		fmt.Printf("Err: %s\n", err)
+	}
+	strSlice, err := rdr.GetList("postgres2")
+	if err != nil {
+		fmt.Printf("Err: %s\n", err)
+	}
+	dataListDeleted := []string{"database1", "database2"}
+	assert.Equal(t, strSlice, dataListDeleted, "A simple Delete operation")
+}
+
+func TestDeleteKey(t *testing.T) {
+	err := wrtr.Store("postgres3", "somedata")
+	if err != nil {
+		fmt.Printf("Err: %s\n", err)
+	}
+	str, err := rdr.Get("postgres3")
+
+	assert.Equal(t, str, "somedata", "Ensure it was stored")
+
+	err = wrtr.Delete("postgres3", "somedata")
```

Comment:
I am kind of confused about how this test passed.

In the implementation of Delete() there is call to only GetList(). But for a single value keys like above, GetList() should return unmarshalling error. So I am not understanding how this test passes.

This ties back to my comments on line 133 about Delete() implementation.